File: src/CommandLineCommands/ConfigCommands/ConfigAddCommand.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;

class ConfigAddCommand : ConfigBaseCommand
{
    public string? Key { get; set; }
    public string? Value { get; set; }

    public override string GetCommandName()
    {
        return "config add";
    }

    public List<Task<int>> Execute(bool interactive)
    {
        var tasks = new List<Task<int>>();
        tasks.Add(Task.FromResult(ExecuteAdd(Key, Value, GetConfigScope())));
        return tasks;
    }

    private int ExecuteAdd(string? key, string? value, ConfigFileScope scope)
    {
        if (string.IsNullOrWhiteSpace(key))
        {
            throw new CommandLineException($"Error: No key specified.");
        }

        if (value == null)
        {
            throw new CommandLineException($"Error: No value specified.");
        }

        _configStore.AddToList(key, value, scope, true);

        var listValue = _configStore.GetFromScope(key, scope).AsList();
        ConsoleHelpers.WriteLine(listValue.Count > 0
            ? $"{key}:\n" + $"- {string.Join("\n- ", listValue)}"
            : $"{key}: (empty list)",
            overrideQuiet: true);

        return 0;
    }
}

File: ts/src/CommandLineCommands/ConfigCommands/ConfigAddCommand.ts
import { ConfigBaseCommand } from './ConfigBaseCommand';
import { ConfigFileScope } from '../../Configuration/ConfigFileScope';
import { KnownSettings } from '../../Configuration/KnownSettings';
import { ConsoleHelpers } from '../../Helpers/ConsoleHelpers';

/**
 * Command to add a value to a list configuration setting.
 */
export class ConfigAddCommand extends ConfigBaseCommand {
  public key?: string;
  public value?: string;

  getCommandName(): string {
    return 'config add';
  }

  isEmpty(): boolean {
    return !this.key || this.key.trim() === '' || this.value === undefined;
  }

  async executeAsync(_interactive: boolean): Promise<number> {
    return await this.executeAdd(
      this.key!,
      this.value!,
      this.scope ?? ConfigFileScope.Local
    );
  }

  private async executeAdd(
    key: string,
    value: string,
    scope: ConfigFileScope
  ): Promise<number> {
    if (!key || key.trim() === '') {
      throw new Error('Error: No key specified.');
    }
    if (value === undefined) {
      throw new Error('Error: No value specified.');
    }

    // Normalize the key if it's a known setting
    let normalizedKey = key;
    if (KnownSettings.isKnown(key)) {
      normalizedKey = KnownSettings.getCanonicalForm(key);
    }

    await this._configStore.addToList(normalizedKey, value, scope);

    // Display the updated list in YAML format
    const updatedValue = await this._configStore.getFromScope(normalizedKey, scope);
    if (updatedValue && Array.isArray(updatedValue.value)) {
      ConsoleHelpers.writeLine(`${normalizedKey}:`);
      for (const item of updatedValue.value) {
        ConsoleHelpers.writeLine(`- ${item}`);
      }
    }

    return 0;
  }

}